### DISCUSSION TITLE
ERROR: DataString is getting too long

### DISCUSSION THREAD

**Discussion by arrabito** [DIRACGrid/DIRAC#6544] [Q&A] [answered]
Dear all,

In our last production we randomly get this kind of error:
 
```
2022-11-03 08:42:55 UTC dirac-jobexec/Script INFO: Command is: /scratch/condor/dir_1381/DIRAC_jk2Ubypilot/19480794/dirac_prod6_muon_run --start_run 0 --run 00000016 --with-half-moon --degraded-mirror-ref MST-NectarCam Paranal muon North 20
2022-11-03 11:08:11 UTC dirac-jobexec/Subprocess ERROR: DataString is getting too long (52428442):  from process 12890 instead of 0.
Got signal 18 from process 12571 instead of 11802.
Got signal 18 from process 12963 instead of 11866.
Got signal 18 from process 12571 instead of 11946.
Got signal 18 from process 12571 instead of 12059.
Got signal 18 from process 12571 instead of 12150.
Got signal 18 from process 12963 instead of 12199.
....

```

We are not able to reproduce this error when running locally and we think it's related to DIRAC.

Could you please tell me if you have an explanation for this kind of issue?

Thank you,

Luisa

**Comment by fstagni** [comment]
This, IIUC, means that the application is printing more than 5MB of output every 0.01 seconds. That's a lot of logging! A quick fix would be that your application is less verbose. 
Otherwise we can increase DIRAC's buffer limit (requires a release) but I am not sure about the possible consequences.

**Reply by andresailer** [reply]
As an in joke to today's BILD: I spent already too much time thinking about getting that error six years ago :/ https://github.com/DIRACGrid/DIRAC/pull/2876/

Limit is set here, I think,
https://github.com/DIRACGrid/DIRAC/blob/f20e8f497c122fd7ecd11dfe9b07dcf8ec33785b/src/DIRAC/Workflow/Modules/Script.py#L40

This can be made configurable with _some_ thinking (but fine for me if you crank that up without thinking about it.)

**Comment by marianne013** [comment]
I think we always told our users that this was a user error, and any logging should go to a file and not the console.

**Comment by fstagni** [comment]
https://github.com/DIRACGrid/DIRAC/pull/6559

**Reply by marianne013** [reply]
You are just encouraging them !!!

**Reply by fstagni** [reply]
Freedom for the people!

**Comment by fstagni** [comment, marked as answer]
So, this issue is in fact already written down in https://github.com/DIRACGrid/DIRAC/issues/5575. I mark this comment as "solution" to this thread.
